src/as2fm/scxml_converter/scxml_entries/scxml_data.py

```diff
@@ -75,8 +75,10 @@ def _interpret_type_from_comment_above(
             return None
         return type_match.group(1), type_match.group(2)
 
-    @staticmethod
-    def from_xml_tree(xml_tree: ET.Element, comment_above: Optional[str] = None) -> "ScxmlData":
+    @classmethod
```

Comment:
I see there is a reason for not using the `from_xml_tree_impl` pattern as for the other classes, but nevertheless you could override the `from_xml_tree` method, and additionally I do not see the reason for switching from the `staticmethod` to the `classmethod`, unless we decide to do this consistently.

Finally, I am wondering if we could take the generalization one step further, by providing an optional `kwargs` style arguments list in `from_xml_tree` and `from_xml_tree_impl`, like explained here: https://www.geeksforgeeks.org/args-kwargs-python/

Reply:
done